In [96]:
import pandas as pd

kaggle_ds = pd.read_csv("data/kaggle_dataset.csv")
data_world_ds = pd.read_csv("data/data_world_dataset.csv").drop(["tweet_id", "author"], axis=1)

In [97]:
kaggle_ds

Text  Emotion
0                                i didnt feel humiliated  sadness
1      i can go from feeling so hopeless to so damned...  sadness
2       im grabbing a minute to post i feel greedy wrong    anger
3      i am ever feeling nostalgic about the fireplac...     love
4                                   i am feeling grouchy    anger
...                                                  ...      ...
21454               Melissa stared at her friend in dism     fear
21455  Successive state elections have seen the gover...     fear
21456               Vincent was irritated but not dismay     fear
21457  Kendall-Hume turned back to face the dismayed ...     fear
21458                    I am dismayed , but not surpris     fear

[21459 rows x 2 columns]

In [98]:
data_world_ds
# enlever des colonnes et les mots qui commencent par @ (dans la tokenization)

sentiment                                            content
0           empty  @tiffanylue i know  i was listenin to bad habi...
1         sadness  Layin n bed with a headache  ughhhh...waitin o...
2         sadness                Funeral ceremony...gloomy friday...
3      enthusiasm               wants to hang out with friends SOON!
4         neutral  @dannycastillo We want to trade with someone w...
...           ...                                                ...
39995     neutral                                   @JohnLloydTaylor
39996        love                     Happy Mothers Day  All my love
39997        love  Happy Mother's Day to all the mommies out ther...
39998   happiness  @niariley WASSUP BEAUTIFUL!!! FOLLOW ME!!  PEE...
39999        love  @mopedronin bullet train from tokyo    the gf ...

[40000 rows x 2 columns]

In [99]:
kaggle_ds["Emotion"].value_counts()

happy       7029
sadness     6265
anger       2993
fear        2652
love        1641
surprise     879
Name: Emotion, dtype: int64

In [100]:
data_world_ds["sentiment"].value_counts()

neutral       8638
worry         8459
happiness     5209
sadness       5165
love          3842
surprise      2187
fun           1776
relief        1526
hate          1323
empty          827
enthusiasm     759
boredom        179
anger          110
Name: sentiment, dtype: int64

Intial ideas :

We should merge together the columns with the same names and happy with happiness.

Some colums **might** go well together, like:
- *sadness* and *empty*
- *enthusiasm* with *happiness*
- *fun* with *happiness*
- *hate* with *anger*


Finally, we should drop the colums that make a too small number of lines.

In [101]:
kaggle_ds.rename({"happy": "happiness"})

Text  Emotion
0                                i didnt feel humiliated  sadness
1      i can go from feeling so hopeless to so damned...  sadness
2       im grabbing a minute to post i feel greedy wrong    anger
3      i am ever feeling nostalgic about the fireplac...     love
4                                   i am feeling grouchy    anger
...                                                  ...      ...
21454               Melissa stared at her friend in dism     fear
21455  Successive state elections have seen the gover...     fear
21456               Vincent was irritated but not dismay     fear
21457  Kendall-Hume turned back to face the dismayed ...     fear
21458                    I am dismayed , but not surpris     fear

[21459 rows x 2 columns]

In [102]:
# use the "neutral" words as stop words ?

neutral       8638
worry         8459 => fear ? sadness ?
happiness     5209 x 
sadness       5165 x 
love          3842 x 
surprise      2187 x 
fun           1776 => happiness ?
relief        1526
hate          1323 => anger ?
empty          827 => sadness ?
enthusiasm     759 => happiness ?
boredom        179
anger          110 x 

# Delete the other columns

In [103]:
data_world_ds["sentiment"].value_counts()

neutral       8638
worry         8459
happiness     5209
sadness       5165
love          3842
surprise      2187
fun           1776
relief        1526
hate          1323
empty          827
enthusiasm     759
boredom        179
anger          110
Name: sentiment, dtype: int64

In [104]:
data_world_rm = data_world_ds[~(data_world_ds["sentiment"].isin(["neutral", "worry", "fun", "relief", "hate", "empty", "enthusiasm", "boredom"]))].copy()

In [94]:
data_world_rm.rename(columns={"sentiment": "Emotion", "content": "Text"}, inplace=True)
merged_and_cleaned_rm = pd.concat([data_world_rm, kaggle_ds], ignore_index=True)
merged_and_cleaned_rm.to_csv("data/merged_and_cleaned_rm.csv", index=False)

# Keep the colums but merge them

In [70]:
data_world_ds["sentiment"].replace(["fun", "enthusiasm"], "happiness", inplace=True)
data_world_ds["sentiment"].replace(["empty"], "sadness", inplace=True)
data_world_ds["sentiment"].replace(["hate"], "anger", inplace=True)
data_world_ds["sentiment"].replace(["worry"], "fear", inplace=True)

In [71]:
data_world_ds["sentiment"].value_counts()

neutral      8638
fear         8459
happiness    7744
sadness      5992
love         3842
surprise     2187
relief       1526
anger        1433
boredom       179
Name: sentiment, dtype: int64

In [72]:
# drop the other columns
data_world_ds = data_world_ds[~(data_world_ds["sentiment"].isin(["neutral", "relief", "boredom"]))].copy()

# Merge the datasets

In [73]:
data_world_ds["sentiment"].value_counts()

fear         8459
happiness    7744
sadness      5992
love         3842
surprise     2187
anger        1433
Name: sentiment, dtype: int64

In [74]:
data_world_ds.shape

(29657, 2)

In [75]:
kaggle_ds.shape

(21459, 2)

In [76]:
data_world_ds.rename(columns={"sentiment": "Emotion", "content": "Text"}, inplace=True)

In [78]:
merged_and_cleaned = pd.concat([data_world_ds, kaggle_ds], ignore_index=True)

In [80]:
merged_and_cleaned

Emotion                                               Text
0        sadness  @tiffanylue i know  i was listenin to bad habi...
1        sadness  Layin n bed with a headache  ughhhh...waitin o...
2        sadness                Funeral ceremony...gloomy friday...
3      happiness               wants to hang out with friends SOON!
4           fear  Re-pinging @ghostridah14: why didn't you go to...
...          ...                                                ...
51111       fear               Melissa stared at her friend in dism
51112       fear  Successive state elections have seen the gover...
51113       fear               Vincent was irritated but not dismay
51114       fear  Kendall-Hume turned back to face the dismayed ...
51115       fear                    I am dismayed , but not surpris

[51116 rows x 2 columns]

In [81]:
merged_and_cleaned.to_csv("data/merged_and_cleaned.csv", index=False)